src/mpyl/utilities/repo/__init__.py

```diff
@@ -134,18 +144,30 @@ def __to_revision(
         intersection = files_in_revision.intersection(files_touched_in_branch)
         return Revision(count, str(revision), intersection)
 
-    def changes_in_branch(self) -> list[Revision]:
-        self.pull_main_branch()
-        revisions = list(
+    def changes_between(self, base_revision: str, head_revision: str) -> list[Commit]:
+        return list(
             reversed(
                 list(
                     self._repo.iter_commits(
-                        f"{self._config.main_branch}..HEAD", no_merges=True
+                        f"{base_revision}..{head_revision}", no_merges=False
```

Comment:
A lot is happening in this pr and I might have missed it, how do you filter out the merges now? I tested it and they are indeed filtered out